### ISSUE TITLE
Add a TargetKind for annotations on imports, exports, and part directives.

### ISSUE THREAD

**Issue by natebosch** [dart-lang/sdk#52274]
This might be blocked on changing `TargetKind` into a class with static constants instead of an enum as discussed in #49796

The recommended pattern for using `@GenerateNiceMocks` from `package:mockito` is to annotate the import of the generated file. This is a pattern we are likely to repeat into other packages.

It would be great if we can statically require the annotation on an import or export. I think it could be fine to have a single `TargetKind.directive`, but it may be more useful to split it up into imports, exports, and parts.

**Comment by matanlurey** [comment]
If you did a bit mask you could do something cute like:

```
static const anyDirective = TargetKind.importDirective | TargetKind.exportDirective | ...
```
